### PR TITLE
feat(rds): add alerting on CA certificates expiration

### PR DESCRIPTION
**Pull request by dcupif**
This commit adds a new alert that will trigger if it detects any instance with a CA certificate with an expiration date scheduled within the next 15 days.

### CODE REVIEW

**Comment by dcupif** [comment]
#sre

File: content/runbooks/rds/RDSCACertificateCloseToExpiration.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    --db-instance-identifier <your_db_instance> \
+    --ca-certificate-identifier <your_new_certificate>
+```
+
+Use the `--apply-immediately` flag if you wish to change the certificate immediately, otherwise it will apply during your next scheduled maintenance window.
+
+Moving away from `rds-ca-2019`, we recommend using the `rds-ca-rsa2048-g1` certificate authority which:
+
+- has the same properties as `rds-ca-2019` (2048 private key, SHA256 signing alg.) so no risk of incompatibility
+- is valid until 2061
```
Comment (vmercierfr) [inline content/runbooks/rds/RDSCACertificateCloseToExpiration.md:54]: I would suggest some style here

```suggestion
{{< hint info >}}
**Tips**

We recommend using the `rds-ca-rsa2048-g1` certificate authority which:

- Has the same properties as `rds-ca-2019` (2048 private key, SHA256 signing alg.) so no risk of incompatibility
- Is valid until 2061
- Change can be done without restarting the instances
{{< /hint >}}
```

File: content/runbooks/rds/RDSCACertificateCloseToExpiration.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+# RDSCACertificateCloseToExpiration
+
+## Meaning
+
+Alert is triggered when an RDS instance is detected using a CA certificate which is going to expire in less than 15 days.
+
+## Impact
+
+If the certificate is not renewed before expiration, all attempts to initiate an SSL/TLS connection to the RDS instance will fail.
+
```
Comment (vmercierfr) [inline content/runbooks/rds/RDSCACertificateCloseToExpiration.md:14]: Majority of the alerts will be related to August 2024 certificate expiration, I would suggest a specific disclamer

```suggestion

{{< hint warning >}}
**Important**

The `Amazon RDS Root 2019 CA` certificate expires on **Aug 22 17:08:50 2024 GMT**.

- Starting January 25th 2024, RDS instances created without specifying the CA will use `rds-ca-rsa2048-g1``.
- In August 2024, AWS will enforce the CA rotation on all RDS instances on the expiring CA during a window maintenance
{{< /hint >}}

```

**Review by vmercierfr — APPROVED** [review]

File: content/runbooks/rds/RDSCACertificateCloseToExpiration.md (hunk 2 above)
Comment (qfritz) [reply content/runbooks/rds/RDSCACertificateCloseToExpiration.md:14]:
```suggestion
**Aug 22 17:08:50 2024 UTC**
```
Small nitpick but I'd recommend using UTC over any local timezone for better internationalisation.

**Review by qfritz — APPROVED** [review]

File: content/runbooks/rds/RDSCACertificateCloseToExpiration.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+      ca_certificate_valid_until: .CertificateDetails.ValidTill
+    } |
+    (now + 1296000) as $date |
+    select (
+      (.ca_certificate_valid_until | split("+")[0] + "Z" | fromdate) < $date
+    )
+  ]'
+```
+
+Note: `1296000` seconds = 15 days
```
[the remark this answers is not in the document]
Comment (dcupif) [reply content/runbooks/rds/RDSCACertificateCloseToExpiration.md]: done 👍

File: content/runbooks/rds/RDSCACertificateCloseToExpiration.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+## Mitigation
+
+- Renew your certificate for the instances retrieved above by running:
```
[the remark this answers is not in the document]
Comment (dcupif) [reply content/runbooks/rds/RDSCACertificateCloseToExpiration.md]: done 👍

File: content/runbooks/rds/RDSCACertificateCloseToExpiration.md (hunk 1 above)
Comment (dcupif) [reply content/runbooks/rds/RDSCACertificateCloseToExpiration.md]: done 👍

File: content/runbooks/rds/RDSCACertificateCloseToExpiration.md (hunk 2 above)
Comment (dcupif) [reply content/runbooks/rds/RDSCACertificateCloseToExpiration.md:14]: done 👍

**Pull request merged**